modules/manage/partials/authentication.adoc

```diff
@@ -0,0 +1,1695 @@
+Authorization (Bearer) header.
+
+The access token is a _bearer token_. A bearer token is used for authentication and
+authorization in web applications and APIs, and holds user credentials, usually in the form
+of random strings of characters. Bearer tokens are generated based on protocols and specifications
+such as JWT (JSON Web Token), which has a header, payload, and signature. The signature must be
+verified according to the JWK. Claims inside the token and the token signature must both be
+validated. After validation, a configurable claim from the token payload is extracted as
+the principal and attached to the connection, as with any other authentication method.
+
+Following is an example JWT header:
+
+[,json]
+----
+{
+  "alg": "RS256",
+  "typ": "JWT",
+  "kid": "tMQzailSAdaW4nojXxES9"
+}
+----
+
+Following is an example JWT payload:
+
+[,json]
+----
+{
+  "iss": "https://dev-ltxchcls4igzho78.us.auth0.com/",
+  "sub": "3JJeI4tmMC6v8mCVCSDnAGVf2vrnJ0BT@clients",
+  "aud": "localhost",
+  "iat": 1694430088,
+  "exp": 1694516488,
+  "azp": "3JJeI4tmMC6v8mCVCSDnAGVf2vrnJ0BT",
+  "scope": "email2",
+  "gty": "client-credentials"
+}
+----
+
+Following are additional validation claims (JWT properties) that are included in the access token:
+
+* `alg`: The signature algorithm. The extension point in the JWT header is the signature algorithm used to sign the token, and cannot contain the value `none`.
+* `aud`:  Audience. Must match the configuration specified in `oidc_token_audience`. Cannot contain the value `none`.
+* `kid`: Key identifier. Must match _any_ of the public JWK listed in the `jwks_uri` endpoint.
+* `exp`: Expiration. The timestamp listed is greater than current time. Must validate within acceptable bounds of the value specified in `oidc_clock_skew_tolerance`. A clock skew tolerance period may be configured by an Admin to account for clock drift between Redpanda and the OIDC Identity Provider (IdP).
+* `iss`: Issuer. Must exactly match the `issuer` property of the JSON returned from the URL specified in `oidc_discovery_url`.
+* `scope`: Scope. Must include the value `openid`.
+* `sub`: Subject. This default claim identifies the principal subject. While `sub` is the default mapping (`$.sub`) in Redpanda, any claim within the JWT can be mapped to a Redpanda principal.
+
+===== Enable OIDC
+
+. Register a client application with your IdP.
++
+A client application, in this context, refers to any application or service that will authenticate against the Redpanda cluster using OIDC.
+This registration process involves creating a new application entry in the IdP's management console.
+During this process, you'll specify details about your application,
+such as the type of application, the callback URLs, and any other required information as per your IdP's requirements.
+In an enterprise environment, OIDC integration typically requires coordination with your organization's security team.
```

Comment:
We should mention and make it very clear that the IDP only interacts with the application by giving it access to endpoints (via discovery url) that it can use to verify tokens. The IDP can then create tokens that are authenticated against the application. It may seem obvious but the fact that the interaction with a token agains the application is the whole point is a bit lost with all the details here.